### PR TITLE
Workaround a dyld issue we are seeing in CI

### PR DESCRIPTION
**Pull request by neonichu**
rdar://88475529

### CODE REVIEW

**Comment by neonichu** [comment]
@swift-ci please test

**Comment by neonichu** [comment]
@swift-ci please test linux

File: Sources/TSCTestSupport/Product.swift [hunk 1]
```diff
@@ -101,6 +101,8 @@ extension Product {
         // Unset the internal env variable that allows skipping certain tests.
         environment["_SWIFTPM_SKIP_TESTS_LIST"] = nil
 
+        environment["DYLD_LIBRARY_PATH"] = nil
```
Comment (tomerd) [inline Sources/TSCTestSupport/Product.swift:104]: is this a temporary measure? seems like this could lead to issues down the line?

**Comment by neonichu** [comment]
@swift-ci please test linux

File: Sources/TSCTestSupport/Product.swift (hunk 1 above)
Comment (neonichu) [reply Sources/TSCTestSupport/Product.swift:104]: Yah, definitely seems temporary to me. This seems to be the underlying issue rdar://75752698

**Pull request closed without merging**